pkg/service/controller_test.go

```diff
@@ -48,24 +49,37 @@ var _ = Describe("StorageClass", func() {
 })
 
 var _ = Describe("CreateVolume", func() {
-	It("should successfully create a volume", func() {
+	DescribeTable("should successfully create a volume", func(cap *csi.VolumeCapability) {
 		client := &ControllerClientMock{}
 		controller := ControllerService{client, testInfraNamespace, testInfraLabels, storageClassEnforcement}
 
-		response, err := controller.CreateVolume(context.TODO(), getCreateVolumeRequest())
+		response, err := controller.CreateVolume(context.TODO(), getCreateVolumeRequest(cap))
```

Comment:
can we validate the DV being created has the expected AccessMode and VolumeMode set as well now